### DISCUSSION TITLE
101 OK response in logs for audio coming from websocket

### DISCUSSION THREAD

**Discussion by sagardesai90** [deepgram/community#418] [General help] [answered]
### Which Deepgram product are you using?

Nova - live transcription 

### Details

I'm sending audio from client using socket.io to node server, and then sending to deepgram for transcription. I'm able to get the audio sent, but am seeing a 101 OK in the logs for requests made. 

Attached screenshots for you to see. 
<img width="1602" alt="Screenshot 2023-11-05 at 5 54 34 PM" src="https://github.com/deepgram/community/assets/24217185/4e2fcf75-b445-4841-b08e-edaba71b70c4">

Client side req to send audio to node server:
<img width="722" alt="Screenshot 2023-11-05 at 5 57 03 PM" src="https://github.com/deepgram/community/assets/24217185/ebd1469e-5d19-4eee-9d9d-6252b26aab38">

Server side code for connecting & reconnecting to deepgram if connection fails, I had to set this up because connection kept failing repeatedly/unexpectedly:
<img width="755" alt="Screenshot 2023-11-05 at 5 58 19 PM" src="https://github.com/deepgram/community/assets/24217185/bba0f470-7c69-442a-96c0-3ae9e7aa7ab7">

Server side code that handles sending of audio received from client, to deepgram. 
<img width="524" alt="Screenshot 2023-11-05 at 5 59 40 PM" src="https://github.com/deepgram/community/assets/24217185/314988c6-7870-4455-8782-174ca89ed701">

Logs from node server, we can see that there is data coming in from the event listener for transcriptReceived, but the contents are undefined:
<img width="446" alt="Screenshot 2023-11-05 at 6 01 36 PM" src="https://github.com/deepgram/community/assets/24217185/75ad01b5-a0cd-4c1a-b5ba-6b7df9abea5a">

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

f4e15645-51f3-458a-a249-14cabbdf090f

### If possible, please attach your code or paste it into the text box.

attached screenshots of the code

### If possible, please attach an example audio file to reproduce the issue.

using live audio from MediaRecorder on client side.

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @sagardesai90, a 101 OK response is expected when streaming audio:

> If the server thinks the HTTP request and any provided configuration options or credentials are valid, it responds with an HTTP 101 code, indicating that the connection is accepted and that the server plans to switch the connection to the WebSocket protocol. After the HTTP 101 response is sent, the client can start to send and receive data from the WebSocket as usual.
([doc link](https://developers.deepgram.com/docs/troubleshooting-websocket-data-and-net-errors-when-live-streaming-audio#establishing-a-websocket-connection))

I see that you're looking for the transcript in Deepgram's response at `data.channel`, and getting an empty transcript. I'd expect that path to work; the text of the transcript will be nested in `data.channel.alternatives.0.transcript`.

I don't see you providing any encoding or sample rate information on your stream. Do you know if you are intending to set that, or if you are streaming containerized audio directly? Is your audio being streamed from a local microphone, or some other source? I wonder if the audio isn't reaching Deepgram in a proper readable format, leading us to successfully receive the audio and send a response, but not decoding the audio and producing the appropriate transcript.

**Reply by sagardesai90** [reply]
I can make some changes to try to reach the text in data.channel.alternatives.0.transcript, and also set the encoding and sample rate info like you recommended. 

Any chance this could be the reason its also closing the connection periodically?

**Reply by sagardesai90** [reply]
I tried to make adjustments like the sampling rate, and encoding, but am still seeing an undefined response.

Also, I am using a local microphone in the browser using MediaRecorder.

**Reply by jkroll-deepgram** [reply]
We have a [tutorial that uses MediaRecorder](https://deepgram.com/learn/how-to-add-speech-recognition-to-your-react-project), maybe consulting that code could help troubleshoot.

We do have a connection timeout of 12 seconds with no audio sent, does that sound like what you're experiencing? You can use our [KeepAlive](https://developers.deepgram.com/reference/streaming#stream-keepalive) feature to keep the websocket alive as long as needed. That includes a `.keepAlive()` method in our [Node SDK](https://github.com/deepgram/deepgram-node-sdk/blob/95f9291f20c92058c013923a357c5a1c6dc7f2de/src/transcription/liveTranscription.ts#L71).

**Reply by aaditya25052002** [reply]
@sagardesai90 were you able to solve the problem later, because I am using the ruby with their APIs, I am also sending all the required parameters, though getting that 101 in response.
p.s. I am getting the audio via twilio call media streams
